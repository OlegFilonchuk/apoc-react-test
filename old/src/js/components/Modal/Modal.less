@modal-header-size: 54px;
@modal-footer-size: 60px;

@modal-margin: 10px;
@modal-tablet-margin: 100px;

@modal-body-spacing: (@modal-header-size + @modal-footer-size + (@modal-margin * 2));
@modal-body-spacing-tablet: (@modal-header-size + @modal-footer-size + (@modal-tablet-margin * 2));

// .modal-dialog class is required in order no to mix styles with the CanJS based version of Apoc library
.modal-dialog .sc-dialog {
  pointer-events: none;

  &,
  &-backdrop {
    &.sc-on {
      display: block;
    }
  }

  &-header {
    height: @modal-header-size;
  }

  &-body {
    overflow-y: auto;
    max-height: calc(100vh - @modal-body-spacing);

    @media (min-width: 768px) {
      max-height: calc(100vh - @modal-body-spacing-tablet);
    }
  }

  &-footer {
    height: @modal-footer-size;
  }

  &-backdrop {
    display: none;

    &:not(.sc-on) {
      z-index: -1;
    }
  }

  .sc-dialog-box {
    pointer-events: all;
    margin: 50px auto;
  }
}
