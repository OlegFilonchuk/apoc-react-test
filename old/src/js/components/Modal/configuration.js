/* eslint-disable import/prefer-default-export */
export const Z_INDEX = 15500;
/* eslint-enable import/prefer-default-export */
