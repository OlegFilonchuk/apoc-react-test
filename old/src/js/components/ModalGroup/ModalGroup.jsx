import React from 'react';
import PropTypes from 'prop-types';

import Backdrop from '../Backdrop/Backdrop';

const getTopZIndex = (currentElementIndex, openedModalNames, zIndex) =>
  zIndex + openedModalNames.length + currentElementIndex;

export default function ModalGroup(props) {
  const getModals = () => {
    const { openedModalNames, zIndex } = props;

    return React.Children.toArray(props.children).map((modal, index) => {
      const zIndexForElement = getTopZIndex(index, openedModalNames, zIndex);
      const key = modal.props.name + zIndex;
      const isOpen = openedModalNames.includes(modal.props.name);

      return React.cloneElement(modal, {
        key,
        isOpen,
        zIndex: zIndexForElement
      });
    });
  };

  const { openedModalNames, zIndex, className } = props;

  const openModalsCount = openedModalNames.length;
  const isBackdropVisible = !!openModalsCount;
  const backdropPosition = openModalsCount - 1;

  const backdropZIndex = getTopZIndex(backdropPosition, openedModalNames, zIndex);
  const modals = getModals();

  return (
    <div className={className}>
      <Backdrop zIndex={backdropZIndex} isVisible={isBackdropVisible} />
      {modals}
    </div>
  );
}

ModalGroup.propTypes = {
  /* eslint-disable react/no-unused-prop-types */
  /**
   * Default z-index for modals and backdrop
   */
  zIndex: PropTypes.number,

  /**
   * Children `<Modal/>` elements
   */
  children: PropTypes.node.isRequired,
  /* eslint-enable react/no-unused-prop-types */

  /**
   * Array of opened modal names
   */
  openedModalNames: PropTypes.arrayOf(PropTypes.string),

  /**
   * Any custom class names
   */
  className: PropTypes.string
};

ModalGroup.defaultProps = {
  zIndex: 15500,
  openedModalNames: [],
  className: ''
};
