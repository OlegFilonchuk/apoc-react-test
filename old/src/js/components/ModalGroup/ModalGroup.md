Purpose of this component is to be used together with [Modal](#modal).

## Example
    const initialState = {
        openModalNames: []
    };

    function openModal(name) {
      const openModalNames = state.openModalNames || [];

      if (openModalNames.includes(name)) {
        return;
      }

      openModalNames.push(name)

      setState({openModalNames});
    }

    function closeModal(name) {
      const openModalNames = state.openModalNames || [];

      if (openModalNames.includes(name)) {
        openModalNames.splice(openModalNames.indexOf(name), 1);
      }

      setState({openModalNames});
    }

    (
    <div>
      <Button className="sc-btn-default" onClick={ openModal.bind(null, 'first_modal') }>Open Modal Dialog</Button>
      <ModalGroup openedModalNames={state.openModalNames}>
        <Modal
          name="first_modal"
          header="Modal Dialog"
          footer={
            <Button onClick={ closeModal.bind(null, 'first_modal') } className="sc-btn-default">
              Close
            </Button>
          }
        >
          Hello World from fist modal!
          A bit larger than you'd expect
          Maybe just to show it's still visible when second is opened
          <Button className="sc-btn-default" onClick={ openModal.bind(null, 'second_modal') } >
            Open Second Modal
          </Button>
        </Modal>
        <Modal
          name="second_modal"
          header="Modal Dialog"
          footer={
            <Button onClick={ closeModal.bind(null, 'second_modal') } className="sc-btn-default">
              Close
            </Button>
          }
        >
          <p>Hello World from second modal!</p>
        </Modal>
      </ModalGroup>
    </div>
    )
