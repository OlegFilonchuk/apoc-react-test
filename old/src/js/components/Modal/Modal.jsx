import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { Z_INDEX } from './configuration';

import './Modal.less';

export default function Modal(props) {
  const dialogClass = classNames('sc-dialog', 'sc-animate', {
    'sc-on': props.isOpen
  });

  return (
    <div className="modal-dialog">
      <div className={dialogClass} style={{ zIndex: props.zIndex }}>
        <div className={`sc-dialog-box ${props.className}`}>
          <div className="sc-dialog-content">
            <div className="sc-dialog-header">
              <h3 className="sc-dialog-title">{props.header}</h3>
            </div>
            <div className="sc-dialog-body">{props.children}</div>
            <div className="sc-dialog-footer">{props.footer}</div>
          </div>
        </div>
      </div>
    </div>
  );
}

Modal.propTypes = {
  /* eslint-disable react/no-unused-prop-types */
  /**
   * Modal name for multistacking
   */
  name: PropTypes.string,
  /* eslint-enable react/no-unused-prop-types */

  /**
   * isOpen determines if modal is opened
   */
  isOpen: PropTypes.bool,

  /**
   * Header content of modal header
   */
  header: PropTypes.node,

  /**
   * Footer content of modal footer
   */
  footer: PropTypes.node,

  /**
   * ClassName size/appearance modifier class
   */
  className: PropTypes.string,

  /**
   * Children modal body content
   */
  children: PropTypes.node,

  /**
   * Modal z-index
   */
  zIndex: PropTypes.number
};

Modal.defaultProps = {
  isOpen: false,
  zIndex: Z_INDEX,
  children: '',
  className: '',
  header: '',
  footer: '',
  name: ''
};
