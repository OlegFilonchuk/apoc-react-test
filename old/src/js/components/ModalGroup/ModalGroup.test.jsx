import React from 'react';
import { shallow } from 'enzyme';

import ModalGroup from './ModalGroup';
import Modal from '../Modal/Modal';

describe('<ModalGroup />', () => {
  const mountModalGroup = (modals, modalsOpen = [], mountFn = shallow) =>
    mountFn(<ModalGroup openedModalNames={modalsOpen}>{modals}</ModalGroup>);

  describe('Modal group render', () => {
    it('should render ModalGroup with one modal', () => {
      const modal = (
        <Modal name="first_modal" header="" footer={<span>test</span>}>
          Hello World from fist modal!
        </Modal>
      );

      const modalGroup = mountModalGroup(modal);

      expect(modalGroup).toMatchSnapshot();
    });

    it('should render ModalGroup with multiple modals', () => {
      const modals = [
        <Modal name="first_modal" key={0} header="" footer={<span>test</span>}>
          Hello World from fist modal!
        </Modal>,
        <Modal name="second_modal" key={1} header="" footer={<span>test2</span>}>
          Hello World from fist modal!
        </Modal>
      ];

      const modalGroup = mountModalGroup(modals);

      expect(modalGroup).toMatchSnapshot();
    });

    it('should render ModalGroup with open modal if modal Id is in multistackModals', () => {
      const openedNames = ['first_modal'];
      const modals = [
        <Modal name="first_modal" key={0} header="" footer={<span>test</span>}>
          Hello World from fist modal!
        </Modal>,
        <Modal name="second_modal" key={1} header="" footer={<span>test2</span>}>
          Hello World from fist modal!
        </Modal>
      ];

      const modalGroup = mountModalGroup(modals, openedNames);

      expect(modalGroup).toMatchSnapshot();
    });
  });
});
