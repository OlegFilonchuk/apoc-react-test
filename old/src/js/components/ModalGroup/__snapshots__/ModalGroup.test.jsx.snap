// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalGroup /> Modal group render should render ModalGroup with multiple modals 1`] = `
<div
  className=""
>
  <Backdrop
    className=""
    isVisible={false}
    onClick={[Function]}
    zIndex={15499}
  />
  <Modal
    className=""
    footer={
      <span>
        test
      </span>
    }
    header=""
    isOpen={false}
    key="first_modal15500"
    name="first_modal"
    zIndex={15500}
  >
    Hello World from fist modal!
  </Modal>
  <Modal
    className=""
    footer={
      <span>
        test2
      </span>
    }
    header=""
    isOpen={false}
    key="second_modal15500"
    name="second_modal"
    zIndex={15501}
  >
    Hello World from fist modal!
  </Modal>
</div>
`;

exports[`<ModalGroup /> Modal group render should render ModalGroup with one modal 1`] = `
<div
  className=""
>
  <Backdrop
    className=""
    isVisible={false}
    onClick={[Function]}
    zIndex={15499}
  />
  <Modal
    className=""
    footer={
      <span>
        test
      </span>
    }
    header=""
    isOpen={false}
    key="first_modal15500"
    name="first_modal"
    zIndex={15500}
  >
    Hello World from fist modal!
  </Modal>
</div>
`;

exports[`<ModalGroup /> Modal group render should render ModalGroup with open modal if modal Id is in multistackModals 1`] = `
<div
  className=""
>
  <Backdrop
    className=""
    isVisible={true}
    onClick={[Function]}
    zIndex={15501}
  />
  <Modal
    className=""
    footer={
      <span>
        test
      </span>
    }
    header=""
    isOpen={true}
    key="first_modal15500"
    name="first_modal"
    zIndex={15501}
  >
    Hello World from fist modal!
  </Modal>
  <Modal
    className=""
    footer={
      <span>
        test2
      </span>
    }
    header=""
    isOpen={false}
    key="second_modal15500"
    name="second_modal"
    zIndex={15502}
  >
    Hello World from fist modal!
  </Modal>
</div>
`;
